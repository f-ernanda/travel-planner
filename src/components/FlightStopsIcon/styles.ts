'use client'

import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    gap: ${theme.grid.gutter};
    justify-content: center;

    > * {
      flex: 0 0 100%;
    }
  `}
`
