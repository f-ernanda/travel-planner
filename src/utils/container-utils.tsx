'use client'

import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Container = styled.div`
  ${({ theme }) => css`
    margin-inline: auto;
    max-width: ${theme.grid.container};
    padding-inline: calc(${theme.grid.gutter} / 2);
    width: 100%;

    ${media.lessThan('medium')`
      display: flex;
      flex-direction: column;
      min-height: 100vh;
    `}
  `}
`
