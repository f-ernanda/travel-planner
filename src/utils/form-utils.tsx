import Joi from 'joi'

const fieldsValidations = {
  email: Joi.string()
    .email({ minDomainSegments: 2, tlds: { allow: ['com', 'net'] } })
    .required(),
  password: Joi.string().required()
}

export type FieldErrors = {
  [key: string]: string
}

function getFieldErrors(objError: Joi.ValidationResult) {
  const errors: FieldErrors = {}

  if (objError.error) {
    objError.error.details.forEach((err) => {
      errors[err.path.join('.')] = err.message
    })
  }

  return errors
}

type SignInValidateParams = {
  email: string
  password: string
}

export function signInValidate(values: SignInValidateParams) {
  const { email, password } = fieldsValidations
  const schema = Joi.object({ email, password })

  return getFieldErrors(schema.validate(values, { abortEarly: false }))
}
