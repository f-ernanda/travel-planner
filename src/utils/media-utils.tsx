'useClient'

import styled, { css } from 'styled-components'
import media, { DefaultBreakpoints } from 'styled-media-query'

type breakpoint = keyof DefaultBreakpoints

type MediaWrapperProps = {
  $lessThan?: breakpoint
  $greaterThan?: breakpoint
}

const mediaWrapperModifiers = {
  lessThan: (size: breakpoint) => css`
    ${media.lessThan(size)`
      display: block;
    `}
  `,

  greaterThan: (size: breakpoint) => css`
    ${media.greaterThan(size)`
      display: block;
    `}
  `
}

export const MediaWrapper = styled.div<MediaWrapperProps>`
  ${({ $lessThan, $greaterThan }) => css`
    display: none;

    ${$lessThan && mediaWrapperModifiers.lessThan($lessThan)}
    ${$greaterThan && mediaWrapperModifiers.greaterThan($greaterThan)}
  `}
`
