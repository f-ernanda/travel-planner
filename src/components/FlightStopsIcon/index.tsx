import { FlightLand, FlightTakeoff } from '@styled-icons/material-rounded'
import * as Styled from './styles'

type FlightStopsIconProps = {
  $stops: number
}

const FlightStopsIcon = ({ $stops }: FlightStopsIconProps) => (
  <Styled.Wrapper>
    <FlightTakeoff />
    {$stops === 0 && (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 267 3"
        aria-hidden
      >
        <path
          stroke="#0D8C84"
          strokeLinecap="round"
          strokeWidth="20"
          d="M2 1.5h263"
        />
      </svg>
    )}
    {$stops === 1 && (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 267 16"
        aria-hidden
      >
        <path
          stroke="#0D8C84"
          strokeLinecap="round"
          strokeWidth="20"
          d="M2 7.5h263"
        />
        <circle cx="133" cy="8" r="30" fill="#0D8C84" />
      </svg>
    )}
    {$stops >= 2 && (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 267 16"
        aria-hidden
      >
        <path
          stroke="#0D8C84"
          strokeLinecap="round"
          strokeWidth="20"
          d="M2 7.5h263"
        />
        <circle cx="107" cy="8" r="30" fill="#0D8C84" />
        <circle cx="160" cy="8" r="30" fill="#0D8C84" />
      </svg>
    )}
    <FlightLand />
  </Styled.Wrapper>
)

export default FlightStopsIcon
