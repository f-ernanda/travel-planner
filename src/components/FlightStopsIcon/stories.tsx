import { Meta, StoryObj } from '@storybook/react'
import FlightStopsIcon from '.'

export default {
  title: 'FlightStopsIcon',
  component: FlightStopsIcon
} as Meta

export const Default: StoryObj = {}
