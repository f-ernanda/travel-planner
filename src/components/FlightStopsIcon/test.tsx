import { renderWithTheme } from '@/utils/test-utils'
import FlightStopsIcon from '.'

describe('<FlightStopsIcon />', () => {
  it('should render the matched icon', () => {
    const { container } = renderWithTheme(<FlightStopsIcon />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the icon correctly when no prop is set', () => {
    const { container } = renderWithTheme(<FlightStopsIcon />)

    expect(container.querySelector('.flight-takeoff')).toBeInTheDocument()
    expect(container.querySelector('.zero-stops-icon')).toBeInTheDocument()
    expect(container.querySelector('.one-stop-icon')).not.toBeInTheDocument()
    expect(
      container.querySelector('.two-or-more-stops-icon')
    ).not.toBeInTheDocument()
    expect(container.querySelector('.flight-land')).toBeInTheDocument()
  })

  it('should render the icon correctly when prop is set with 1 stops', () => {
    const { container } = renderWithTheme(<FlightStopsIcon $stops={1} />)

    expect(container.querySelector('.flight-takeoff')).toBeInTheDocument()
    expect(container.querySelector('.one-stop-icon')).toBeInTheDocument()
    expect(container.querySelector('.flight-land')).toBeInTheDocument()
  })

  it('should render the icon correctly when prop is set with 2 stops', () => {
    const { container } = renderWithTheme(<FlightStopsIcon $stops={2} />)

    expect(container.querySelector('.flight-takeoff')).toBeInTheDocument()
    expect(
      container.querySelector('.two-or-more-stops-icon')
    ).toBeInTheDocument()
    expect(container.querySelector('.flight-land')).toBeInTheDocument()
  })
})
