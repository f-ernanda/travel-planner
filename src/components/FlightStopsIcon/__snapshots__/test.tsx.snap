// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlightStopsIcon /> should render the matched icon 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: flex;
  gap: 2.8rem;
  justify-content: center;
}

.c0 >* {
  flex: 0 0 100%;
}

<div
  class="c0"
>
  <svg
    aria-hidden="true"
    class="c1 flight-takeoff"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.5 19h-17c-.55 0-1 .45-1 1s.45 1 1 1h17c.55 0 1-.45 1-1s-.45-1-1-1zm1.57-9.36c-.22-.8-1.04-1.27-1.84-1.06L14.92 10 8.46 3.98a1.06 1.06 0 0 0-1.02-.25c-.68.19-1 .97-.65 1.58l3.44 5.96-4.97 1.33-1.57-1.24c-.25-.19-.57-.26-.88-.18l-.33.09c-.32.08-.47.45-.3.73l1.88 3.25c.23.39.69.58 1.12.47L21 11.48c.8-.22 1.28-1.04 1.07-1.84z"
    />
  </svg>
  <svg
    aria-hidden="true"
    class="zero-stops-icon"
    fill="none"
    viewBox="0 0 267 3"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 1.5h263"
      stroke="#0D8C84"
      stroke-linecap="round"
      stroke-width="20"
    />
  </svg>
  <svg
    aria-hidden="true"
    class="c1 flight-land"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.5 19h-17c-.55 0-1 .45-1 1s.45 1 1 1h17c.55 0 1-.45 1-1s-.45-1-1-1zM3.51 11.61l15.83 4.24c.8.21 1.62-.26 1.84-1.06.21-.8-.26-1.62-1.06-1.84l-5.31-1.42-2.58-8.45a1.08 1.08 0 0 0-.75-.73c-.68-.18-1.35.33-1.35 1.04v6.88L5.15 8.95 4.4 7.09a.991.991 0 0 0-.67-.59l-.33-.09a.495.495 0 0 0-.63.48v3.75c0 .46.3.85.74.97z"
    />
  </svg>
</div>
`;
